server/WebAPI/Controllers/AuthController.cs

```diff
@@ -13,25 +15,50 @@ public class AuthController : Controller
   {
     private readonly UserService _userService;
     private readonly AuthService _authService;
+    private readonly GoogleOauthService _googleOauthService;
 
-    public AuthController(UserService userService, AuthService authService)
+    public AuthController(UserService userService, AuthService authService, GoogleOauthService googleOauthService)
     {
       _userService = userService;
       _authService = authService;
+      _googleOauthService = googleOauthService;
     }
 
     [AllowAnonymous]
     [HttpPost]
     public IActionResult Auth(AuthRequest authRequest)
     {
       var user = _userService.GetUser(authRequest.Email, authRequest.Password);
-      if(user == null)
+      if (user == null)
         return Unauthorized();
 
       var token = _authService.MakeToken(user);
       return Ok(new AuthResponse { AccessToken = token });
     }
 
+    [AllowAnonymous]
+    [HttpPost("google")]
+    public async Task<IActionResult> AuthWithGoogle(GoogleOauthDTO gAuth)
+    {
+      string googleBearer = await _googleOauthService.GetToken(gAuth.Code);
+      var emailUserId = await _googleOauthService.GetGoogleUserData(googleBearer);
+      
+      var user = _userService.GetUser(emailUserId.Key);
+      if (emailUserId.Key == null)
```

Comment:
why are we checking for key to be null after we're getting user?

[desirable] - check "key" for null before getting user